multiarch/trigger-pipeline.groovy

```diff
@@ -54,7 +54,14 @@ node(vars.node(env.ACT_ON_ARCH, 'trigger')) {
 		if (0 != sh(returnStatus: true, script: '''#!/usr/bin/env bash
 			set -Eeuo pipefail
 			set -x
-			commit="$(wget -qO- "https://doi-janky.infosiftr.net/job/multiarch/job/$ACT_ON_ARCH/job/$REPO/lastSuccessfulBuild/artifact/build-info/commit.txt")"
+
+			# ipv6 can be extremely slow on s390x so test a single ping and if it can't respond in 5 seconds, then swap the wget to use ipv4
+			wgetArgs=()
+			if ! ping -c1 -W5 -6 doi-janky.infosiftr.net &> /dev/null; then
+				wgetArgs+=( '--inet4-only' )
+			fi
+
+			commit="$(wget "${wgetArgs[@]}" -qO- "https://doi-janky.infosiftr.net/job/multiarch/job/$ACT_ON_ARCH/job/$REPO/lastSuccessfulBuild/artifact/build-info/commit.txt")"
```

Comment:
🤔 A slight race condition if the `ping` is successful and then the network is slow during `wget`, but still better than what we have.

Reply:
I think we could do something interesting with `wget --timeout` instead to help mitigate that (and to mitigate the case where the network slowness doesn't exhibit during `ping`, because that might be just our _current_ symptoms not the general network flakiness).  Maybe something like `if ! commit="$(wget --timeout=2 ...)"; then commit="$(wget --inet4-only --timeout=2 ...)"` ? (or maybe a slightly more generous timeout on the second round?)

Reply:
That does seem a little better, though I am not sure about timeout values.
```bash
# ipv6 can be extremely slow on s390x so set a timeout and if it can't respond in 5 seconds, then swap the wget to use ipv4
wgetArgs=( '-qO-' "https://doi-janky.infosiftr.net/job/multiarch/job/$ACT_ON_ARCH/job/$REPO/lastSuccessfulBuild/artifact/build-info/commit.txt" )
if ! commit="$(wget --timeout=5 "${wgetArgs[@]}")"; then
	commit="$(wget --inet4-only --timeout=5 "${wgetArgs[@]}")"
fi
```

Reply:
10 seconds total seems pretty sane -- if our server isn't responding within 5 seconds, we really need to fix that :see_no_evil:

Reply:
Any value south of two minutes will help with our current problem :joy:

Reply:
Ah, just adding `--timeout=5` makes it fast and doesn't fallback to the second `wget` because `wget` will try the next addresses in the DNS response (so the other IPv6 and then the IPv4 addresses).